frontend-runitup/src/App.jsx

```diff
@@ -108,19 +109,19 @@ function App() {
           <Route path="/signup" element={<SignUp />} />
           <Route
             path="/feed"
-            element={user ? <Feed user={user} onLogout={handleLogout} /> : <Navigate to="/login" />}
+            element={user ? <Feed user={user} /> : <Navigate to="/login" />}
```

Comment:
For all of these, I'd suggest restructuring it so that you don't need to include so much user ? <Some page> : navigate to login. There's a lot of duplicate logic here. Instead, use the same `isLoggedIn` boolean constant. If it's false, you can navigate to/render login page. Otherwise, you can show the other pages (feed, routes, recommendation etc)

Reply:
Creation of ProtectedRoute component:
const ProtectedRoute = ({ children, isLoggedIn }) => {
  return isLoggedIn ? children : <Navigate to="/login" />;
};
This is a new component that wraps routes requiring authentication. It renders its children if the user is logged in, otherwise it redirects to the login page. Protected routes now use the ProtectedRoute component instead of inline conditional rendering. This makes the route definitions cleaner and more consistent. It also centralizes the logic for redirecting unauthenticated users.